guidelines/enhancement_template.md

```diff
@@ -461,6 +462,19 @@ Describe how to
   - Namespaces deletion will not delete all objects in etcd, leading to zombie
     objects when another namespace with the same name is created.
 
+#### Operating at scale
+
+It is of special relevance the declarative nature of APIs when operating
+features on multiple clusters. If applicable, reason about any peculiarity to
+the API that does not follow this principle and thus may require specific
+considerations when operated at scale.
+
+For example, the workflow to operate the feature may comprise of several steps
+of different nature instead of a single declarative express of intention. A
+service reliability engineer would prefer for this not to be the case but if it
+must be, he might require additional knowledge to enable the automation of those
```

Comment:
If there's an underlying assumption that automation is possible, why is it not considered as a part of the feature change?  If the steps require customer to take more than one action and/or could cause service degredation it's work a sync with SD on what those implications are.  Ideal state, we do not build something that cannot be used as intended.  To that end, more engagement up front such as the other note in this PR to include other parties like SD is important.

Reply:
I guess this is what this comment is trying to say.

First, that if at all possible, the feature should be automated through a single declarative express of intention.
Second, that if that's is not possible, engagement with SD teams might be required to look into the implications of not doing so.

Perhaps it could be expressed better, but that is what I am trying to say here.